File: Contribute/Plugin Samples/Plugin Samples/Ruby/QuickNotePlugin/MainForm.rb
require "mscorlib"
require "System, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.IO, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Collections.Generic, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.ComponentModel, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Data, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Drawing, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Linq, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Text, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "System.Xml.Linq, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "SS.Ynote.Classic, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "SS.Ynote.Engine.Framework.Plugins, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "SS.Ynote.Engine.Framework.Plugins.Controls, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "SS.Ynote.Engine.Framework.Plugins.Interface, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"
require "WeifenLuo.WinFormsUI.Docking, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a"

module QuickNotePlugin
	class MainForm < DockContent, IFormPlugin
		def initialize()
			@configuration = XElement.new("ThisFormConfig")
			self.InitializeComponent()
			self.@newToolStripMenuItem1.Click { |sender, e| self.menuItem2_Click(sender, e) }
			self.@openToolStripMenuItem1.Click { |sender, e| self.menuItem3_Click(sender, e) }
			self.@importFromMainEditorToolStripMenuItem.Click { |sender, e| self.menuItem18_Click(sender, e) }
			self.@saveAsToolStripMenuItem1.Click { |sender, e| self.menuItem5_Click(sender, e) }
			self.@clearAllToolStripMenuItem.Click { |sender, e| self.menuItem2_Click(sender, e) }
			self.@cutToolStripMenuItem.Click { |sender, e| self.menuItem8_Click(sender, e) }
			self.@copyToolStripMenuItem.Click { |sender, e| self.menuItem9_Click(sender, e) }
			self.@pasteToolStripMenuItem.Click { |sender, e| self.menuItem10_Click(sender, e) }
			self.@selectAllToolStripMenuItem.Click { |sender, e| self.menuItem11_Click(sender, e) }
			self.@undoToolStripMenuItem.Click { |sender, e| self.menuItem7_Click(sender, e) }
			self.@fontToolStripMenuItem.Click { |sender, e| self.menuItem14_Click(sender, e) }
			self.@backColorToolStripMenuItem.Click { |sender, e| self.menuItem15_Click(sender, e) }
		end

		def Content
			return self
		end

		def ShowAs
			return self.ShowAs.Normal
		end

		def Title
			return "QuickNotePlugin"
		end

		def Description
			return "This is  a Quick Note Plugin for Ynote Classic"
		end

		def Group
			return "Plugins"
		end

		def SubGroup
			return "Note"
		end

		def Configuration
			return @configuration
		end

		def Configuration=(value)
			@configuration = value
		end

		def Icon
			#get { return "C:\\Icons\\Folder.ico"; }
			return System::String.Empty
		end

		def menuItem7_Click(sender, e)
			self.@txtmain.Undo()
		end

		def menuItem18_Click(sender, e)
			self.@txtmain.Text = Main.UserText
		end

		def menuItem2_Click(sender, e)
			self.@txtmain.Clear()
		end

		def menuItem3_Click(sender, e)
			o = System.Windows.Forms.OpenFileDialog.new()
			o.Filter = "Text Files (*.txt)|*.txt|All Files(*.*)|*.*"
			o.ShowDialog()
			txtmain.Text = File.ReadAllText(o.FileName, Encoding.Default)
		end

		def menuItem8_Click(sender, e)
			self.@txtmain.Cut()
		end

		def menuItem9_Click(sender, e)
			self.@txtmain.Copy()
		end

		def menuItem10_Click(sender, e)
			self.@txtmain.Paste()
		end

		def menuItem11_Click(sender, e)
			self.@txtmain.SelectAll()
		end

		def menuItem14_Click(sender, e)
			f = System.Windows.Forms.FontDialog.new()
			f.ShowEffects = true
			f.ShowColor = true
			f.ShowDialog()
			txtmain.Font = f.Font
			txtmain.ForeColor = f.Color
		end

		def menuItem15_Click(sender, e)
			c = System.Windows.Forms.ColorDialog.new()
			c.ShowDialog()
			self.@txtmain.BackColor = c.Color
		end

		def menuItem5_Click(sender, e)
			s = System.Windows.Forms.SaveFileDialog.new()
			s.Filter = "All Files (*.*)|*.*"
			s.ShowDialog()
			if s.FileName != "" then
				File.WriteAllText(s.FileName, txtmain.Text, Encoding.Default)
			end
		end

		def menuItem19_Click(sender, e)
			self.Close()
		end

		def exitToolStripMenuItem1_Click(sender, e)
			Application.Exit()
		end
	end
end

File: Contribute/Plugin Samples/IronPython/QuickNotePlugin/MainForm.py
from System import *
from System.IO import *
from System.Collections.Generic import *
from System.ComponentModel import *
from System.Data import *
from System.Drawing import *
from System.Linq import *
from System.Text import *
from System.Windows.Forms import *
from System.Xml.Linq import *
from SS.Ynote.Classic import *
from SS.Ynote.Engine.Framework.Plugins import *
from SS.Ynote.Engine.Framework.Plugins.Controls import *
from SS.Ynote.Engine.Framework.Plugins.Interface import *
from WeifenLuo.WinFormsUI.Docking import *

class MainForm(DockContent, IFormPlugin):
	def __init__(self):
		self._configuration = XElement("ThisFormConfig")
		self.InitializeComponent()
		self._newToolStripMenuItem1.Click += self.menuItem2_Click
		self._openToolStripMenuItem1.Click += self.menuItem3_Click
		self._importFromMainEditorToolStripMenuItem.Click += self.menuItem18_Click
		self._saveAsToolStripMenuItem1.Click += self.menuItem5_Click
		self._clearAllToolStripMenuItem.Click += self.menuItem2_Click
		self._cutToolStripMenuItem.Click += self.menuItem8_Click
		self._copyToolStripMenuItem.Click += self.menuItem9_Click
		self._pasteToolStripMenuItem.Click += self.menuItem10_Click
		self._selectAllToolStripMenuItem.Click += self.menuItem11_Click
		self._undoToolStripMenuItem.Click += self.menuItem7_Click
		self._fontToolStripMenuItem.Click += self.menuItem14_Click
		self._backColorToolStripMenuItem.Click += self.menuItem15_Click

	def get_Content(self):
		return self

	Content = property(fget=get_Content)

	def get_ShowAs(self):
		return self.ShowAs.Normal

	ShowAs = property(fget=get_ShowAs)

	def get_Title(self):
		return "QuickNotePlugin"

	Title = property(fget=get_Title)

	def get_Description(self):
		return "This is  a Quick Note Plugin for Ynote Classic"

	Description = property(fget=get_Description)

	def get_Group(self):
		return "Plugins"

	Group = property(fget=get_Group)

	def get_SubGroup(self):
		return "Note"

	SubGroup = property(fget=get_SubGroup)

	def get_Configuration(self):
		return self._configuration

	def set_Configuration(self, value):
		self._configuration = value

	Configuration = property(fget=get_Configuration, fset=set_Configuration)

	def get_Icon(self):
		#get { return "C:\\Icons\\Folder.ico"; }
		return str.Empty

	Icon = property(fget=get_Icon)

	def menuItem7_Click(self, sender, e):
		self._txtmain.Undo()

	def menuItem18_Click(self, sender, e):
		self._txtmain.Text = Main.UserText

	def menuItem2_Click(self, sender, e):
		self._txtmain.Clear()

	def menuItem3_Click(self, sender, e):
		o = System.Windows.Forms.OpenFileDialog()
		o.Filter = "Text Files (*.txt)|*.txt|All Files(*.*)|*.*"
		o.ShowDialog()
		txtmain.Text = File.ReadAllText(o.FileName, Encoding.Default)

	def menuItem8_Click(self, sender, e):
		self._txtmain.Cut()

	def menuItem9_Click(self, sender, e):
		self._txtmain.Copy()

	def menuItem10_Click(self, sender, e):
		self._txtmain.Paste()

	def menuItem11_Click(self, sender, e):
		self._txtmain.SelectAll()

	def menuItem14_Click(self, sender, e):
		f = System.Windows.Forms.FontDialog()
		f.ShowEffects = True
		f.ShowColor = True
		f.ShowDialog()
		txtmain.Font = f.Font
		txtmain.ForeColor = f.Color

	def menuItem15_Click(self, sender, e):
		c = System.Windows.Forms.ColorDialog()
		c.ShowDialog()
		self._txtmain.BackColor = c.Color

	def menuItem5_Click(self, sender, e):
		s = System.Windows.Forms.SaveFileDialog()
		s.Filter = "All Files (*.*)|*.*"
		s.ShowDialog()
		if s.FileName != "":
			File.WriteAllText(s.FileName, txtmain.Text, Encoding.Default)

	def menuItem19_Click(self, sender, e):
		self.Close()

	def exitToolStripMenuItem1_Click(self, sender, e):
		Application.Exit()